### PR TITLE
Update the edit scene root in time after it is replaced

### LINKED ISSUE #68575: Changing parent node for FileDialog when it's visible will block any input in the editor
### Godot version

4.0.beta4

### System information

Windows 10 21H2, Vulkan, AMD Radeon RX 6750 XT (31.0.12029.1042)

### Issue description

Changing the type of the parent node for the FileDialog when the FileDialog is set to be visible in the Editor will block any input for the Editor (mouse, keyboard inputs as well as close button).

[Video showing the problem](https://www.youtube.com/watch?v=SXR0uSBrj7I)

This also happens for other nodes, for example AcceptDialog

It does not happen in Godot 3.5.1

### Steps to reproduce

1. Create a scene with Node2D
2. Add a FileDialog node as a child to Node2D
3. Change the visibility of the FileDialog node so it is visible in the editor:
obraz
4. Change the type of the Node2D to Control

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Rindbee**
When changing the type of the edited scene root node, it may be necessary to update edited scene root in time for the child nodes to work properly.

Fix #68575.

### CODE REVIEW

**Comment by Rindbee** [comment]
As requested in the documentation, use a signal implementation instead. See [Editor dependencies in scene/ files](https://docs.godotengine.org/en/latest/development/editor/introduction_to_editor_development.html#editor-dependencies-in-scene-files).

File: doc/classes/Node.xml [hunk 1]
```diff
@@ -809,6 +809,13 @@
 				Emitted when the node is renamed.
 			</description>
 		</signal>
+		<signal name="replacing_by">
+			<param index="0" name="node" type="Node" />
+			<description>
+				Emitted when this node is replacing by the [param node], see [method replace_by].
```
Comment (KoBeWi) [inline doc/classes/Node.xml:815]:
```suggestion
				Emitted when this node is being replaced by the [param node], see [method replace_by].
```

Comment (Rindbee) [reply doc/classes/Node.xml]: Thanks for the correction.

**Review by KoBeWi — APPROVED** [review]
Appears to be working correctly.

File: doc/classes/Node.xml [hunk 2]
```diff
@@ -845,6 +845,13 @@
 				Emitted when the node is renamed.
 			</description>
 		</signal>
+		<signal name="replacing_by">
+			<param index="0" name="node" type="Node" />
+			<description>
+				Emitted when this node is being replaced by the [param node], see [method replace_by].
+				This signal is emitted [i]after[/i] [param node] has been added as a child of the original parent node, but [i]before[/i] all original child nodes have been added to [param node].
```
Comment (YuriSizov) [inline doc/classes/Node.xml:852]:
```suggestion
				This signal is emitted [i]after[/i] [param node] has been added as a child of the original parent node, but [i]before[/i] all original child nodes have been reparented to [param node].
```

I think that would be a little bit clearer.

**Review by YuriSizov — APPROVED** [review]

File: doc/classes/Node.xml (hunk 2 above)
Comment (Rindbee) [reply doc/classes/Node.xml]: Changed as suggested.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**